models/spring-ai-ollama/src/main/java/org/springframework/ai/ollama/api/OllamaModel.java

```diff
@@ -30,118 +30,159 @@ public enum OllamaModel implements ChatModelDescription {
 
 	/**
 	 * The 2.7B uncensored Dolphin model
 	 */
-	DOLPHIN_PHI("dolphin-phi"),
+	DOLPHIN_PHI("dolphin-phi", null),
 
 	/**
 	 * The Phi-2 2.7B language model
 	 */
-	PHI("phi"),
+	PHI("phi", null),
 
 	/**
 	 * The Phi-3 3.8B language model
 	 */
-	PHI3("phi3"),
+	PHI3("phi3", null),
 
 	/**
 	 * A fine-tuned Mistral model
 	 */
-	NEURAL_CHAT("neural-chat"),
+	NEURAL_CHAT("neural-chat", null),
 
 	/**
 	 * Starling-7B model
 	 */
-	STARLING_LM("starling-lm"),
+	STARLING_LM("starling-lm", null),
 
 	/**
 	 * Code Llama is based on Llama 2 model
 	 */
-	CODELLAMA("codellama"),
+	CODELLAMA("codellama", null),
 
 	/**
 	 * Orca Mini is based on Llama and Llama 2 ranging from 3 billion parameters to 70
 	 * billion
 	 */
-	ORCA_MINI("orca-mini"),
+	ORCA_MINI("orca-mini", null),
 
 	/**
 	 * Llava is a Large Language and Vision Assistant model
 	 */
-	LLAVA("llava"),
+	LLAVA("llava", null),
 
 	/**
 	 * Gemma is a lightweight model with 2 billion and 7 billion
 	 */
-	GEMMA("gemma"),
+	GEMMA("gemma", null),
 
 	/**
 	 * Uncensored Llama 2 model
 	 */
-	LLAMA2_UNCENSORED("llama2-uncensored"),
+	LLAMA2_UNCENSORED("llama2-uncensored", null),
 
 	/**
 	 * A high-performing open embedding model with a large token context window.
 	 */
-	NOMIC_EMBED_TEXT("nomic-embed-text"),
+	NOMIC_EMBED_TEXT("nomic-embed-text", null),
 
 	/**
 	 * State-of-the-art large embedding model from mixedbread.ai
 	 */
-	MXBAI_EMBED_LARGE("mxbai-embed-large");
+	MXBAI_EMBED_LARGE("mxbai-embed-large", null),
+	/**
+	 * Qwen2.5 models are pretrained on Alibaba's latest large-scale dataset, encompassing
+	 * up to 18 trillion tokens. The model supports up to 128K tokens and has multilingual
+	 * support.
+	 */
+	QWEN2_5("qwen2.5", null), QWEN2_5_7b("qwen2.5", "7b"), QWEN2_5_14b("qwen2.5", "14b"), QWEN2_5_32b("qwen2.5", "32b"),
+	QWEN2_5_72b("qwen2.5", "72b"),
+	/**
+	 * The latest series of Code-Specific Qwen models, with significant improvements in
+	 * code generation, code reasoning, and code fixing.
+	 */
+	QWEN2_5_CODER("qwen2.5-coder", null), QWEN2_5_CODER_3b("qwen2.5-coder", "3b"),
+	QWEN2_5_CODER_7b("qwen2.5-coder", "7b"), QWEN2_5_CODER_14b("qwen2.5-coder", "14b"),
+	QWEN2_5_CODER_32b("qwen2.5-coder", "32b"),
+	/**
+	 * Llama 3.2 Vision is a collection of instruction-tuned image reasoning generative
+	 * models in 11B and 90B sizes.
+	 */
+	LLAMA3_2_VISION("llama3.2-vision", null), LLAMA3_2_VISION_11b("llama3.2-vision", "11b"),
+	LLAMA3_2_VISION_90b("llama3.2-vision", "90b");
 
+	/**
+	 * model id
+	 */
 	private final String id;
 
-	OllamaModel(String id) {
+	/**
+	 * model size
+	 */
+	private final String size;
+
+	OllamaModel(String id, String size) {
 		this.id = id;
+		this.size = size;
 	}
 
 	public String id() {
 		return this.id;
 	}
 
+	/**
+	 * for example: qwen2.5:7b,llama3.2:1b
+	 * @return model id with size
+	 */
 	@Override
 	public String getName() {
-		return this.id;
+		return this.id + (null == this.size ? "" : ":" + this.size);
```

Comment:
Missing Null Handling in getName:

The logic in getName assumes size will be null or non-empty but doesn't guard against unexpected input like an empty string.
Add validation to handle this edge case

Reply:
If there is no difference between id and name, then that's right